python/cuml/preprocessing/model_selection.py

```diff
@@ -14,46 +14,42 @@
 #
 
 import cudf
-from typing import Union, Tuple
-from numba import jit
+import cupy as cp
 import numpy as np
 
-
-@jit
-def _shuffle_idx(idx: np.ndarray):
-    """ Shuffle idx in place which will be used as indices to split a
-    dataframe of size len(np.ndarray)
-    """
-    # TODO this is the bottleneck and should be a gpu operation,
-    # when possible replace with the mlprim mentioned in cuml #659
-    np.random.shuffle(idx)
+from cuml.utils.cupy_utils import test_numba_cupy_version_conflict
+from cuml.utils.numba_utils import PatchedNumbaDeviceArray
+from numba import cuda
+from typing import Union
 
 
 def train_test_split(
-    X: cudf.DataFrame,
-    y: Union[str, cudf.Series],
-    train_size: Union[float, int] = 0.8,
+    X,
+    y,
+    test_size: Union[float, int] = None,
+    train_size: Union[float, int] = None,
     shuffle: bool = True,
-    seed: int = None,
-) -> Tuple[cudf.DataFrame, cudf.DataFrame, cudf.DataFrame, cudf.DataFrame]:
+    seed: Union[int, cp.random.RandomState, np.random.RandomState] = None
+):
     """
-    Partitions the data into four collated dataframes, mimicing sklearn's
-    `train_test_split`
+    Partitions device data into four collated objects, mimicking
+    Scikit-learn's `train_test_split`
 
     Parameters
     ----------
-    X : cudf.DataFrame
+    X : cudf.DataFrame or cuda_array_interface compliant device array
```

Comment:
Should this also support non-gpu arrays? (Not making a new requirement for this PR, just asking for your thoughts).

Reply:
I don't think so, there is no advantage of using cuML for that over Scikit if the interest is splitting host arrays I think